pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -392,6 +392,9 @@ def cache_result(self) -> None:
             # attach capture batch_size
             Result.attach_batch_size(self._batch_size, hook_result)
 
+            hook_result.detach()
+            hook_result.cpu()
```

Comment:
Both detach and cpu aren't inplace. For them to take effect you need to assign back to variable or the result is lost
```
hook_result = hook_result.detach().cpu()
```
Am I missing something?

Reply:
```
    def detach(self):
        for k, v in self.items():
            if isinstance(v, torch.Tensor):
                self.__setitem__(k, v.detach())

    def cpu(self):
        for k, v in self.items():
            if isinstance(v, torch.Tensor):
                self.__setitem__(k, v.cpu())
```

They are inplace operation :)

Reply:
oh, maybe you should follow torch convention and name inplace operations with _ in the end.
like detach_ in torch https://pytorch.org/docs/stable/tensors.html